recipe-server/normandy/recipes/api/v2/shield_identicon.py

```diff
@@ -0,0 +1,81 @@
+from django.template.loader import render_to_string
+
+
+def pick_pair(genome, base_color):
+    good_pairs = [
+        color for color in genome.colors if base_color.contrasts_well(color)]
+    assert len(good_pairs), 'no colors satisfy the luminance requirement'
+    return genome.choice(good_pairs)
+
+
+def single_color(genome, context):
+    context['treatment'] = 'SingleColor'
+    return context
+
+
+def two_color(genome, context):
+    context['treatment'] = 'TwoColor'
+    areas = [
+        'top',
+        'bottom',
+        'left',
+        'right',
+        'topLeft',
+        'topRight',
+        'bottomLeft',
+        'bottomRight',
+    ]
+    area = genome.choice(areas)
```

Comment:
You're not using the names anywhere, could you just use `choice` on a list of the values from the `rotations` dict? You could even just generate it with the `range` builtin:

```python
range(0, 360, 45)  # == an iterator containing the angles you want
```